File: benchmark/php/dfs.php
<?php
$n = 100;
$adj = [];
for ($i = 0; $i < $n; $i++) {
    $row = [];
    for ($j = 0; $j < $n; $j++) {
        $row[] = (($i * 3 + $j * 7) % 11 == 0 && $i != $j) ? 1 : 0;
    }
    $adj[] = $row;
}
$start = hrtime(true);
$visited = array_fill(0, $n, false);
$stack = [0];
$count = 0;
while (count($stack) > 0) {
    $node = array_pop($stack);
    if (!$visited[$node]) {
        $visited[$node] = true;
        $count++;
        for ($neighbor = $n - 1; $neighbor >= 0; $neighbor--) {
            if ($adj[$node][$neighbor] == 1 && !$visited[$neighbor]) {
                $stack[] = $neighbor;
            }
        }
    }
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Visited: " . $count . "
";
echo "Time: " . round($ms) . "ms
";

File: benchmark/python/dfs.py
import time

n = 100
adj = []
for i in range(n):
    row = []
    for j in range(n):
        row.append(0)
    adj.append(row)
for i in range(n):
    for j in range(n):
        if (i * 3 + j * 7) % 11 == 0:
            if i != j:
                adj[i][j] = 1
start = time.time() * 1000
visited = []
for i in range(n):
    visited.append(False)
stack = [0]
count = 0
while True:
    if len(stack) == 0:
        break
    node = stack.pop()
    if not visited[node]:
        visited[node] = True
        count = count + 1
        for neighbor in range(n - 1, -1, -1):
            if adj[node][neighbor] == 1:
                if not visited[neighbor]:
                    stack.append(neighbor)
end = time.time() * 1000
print(f"Visited: {count}")
print(f"Time: {end - start:.0f}ms")

